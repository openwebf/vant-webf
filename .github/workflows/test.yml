name: CI

on:
  push:
    branches:
      - '**'

  pull_request:
    branches:
      - dev

  workflow_dispatch:

jobs:
  lint:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2

    - name: Install pnpm
      run: npm i pnpm@7 -g

    - uses: actions/setup-node@v2
      with:
        node-version: '16'
        cache: 'pnpm'

    - name: Install dependencies
      run: pnpm install

    - name: Run linter
      run: npm run lint

  test:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2

    - name: Install pnpm
      run: npm i pnpm@7 -g

    - uses: actions/setup-node@v2
      with:
        node-version: '16'
        cache: 'pnpm'

    - name: Install dependencies
      run: pnpm install

    - name: Run test cases
      run: npm test

    - name: Upload coverage to Codecov  
      uses: codecov/codecov-action@v1
      with:
        token: ${{ secrets.CODECOV_TOKEN }}

  build:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2

    - name: Install pnpm
      run: npm i pnpm@7 -g

    - uses: actions/setup-node@v2
      with:
        node-version: '16'
        cache: 'pnpm'

    - name: Install dependencies
      run: pnpm install

    - name: Build
      run: npm run build
