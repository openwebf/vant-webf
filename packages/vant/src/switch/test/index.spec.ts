import { Switch } from '..';
import { mount } from '../../../test';

test('should emit update:modelValue event when click the switch button', async () => {
  const wrapper = mount(Switch);

  wrapper.trigger('click');
  expect(wrapper.emitted('update:modelValue')).toHaveLength(1);
  expect(wrapper.emitted('update:modelValue')![0]).toEqual([true]);

  await wrapper.setProps({ modelValue: true });
  wrapper.trigger('click');
  expect(wrapper.emitted('update:modelValue')).toHaveLength(2);
  expect(wrapper.emitted('update:modelValue')![1]).toEqual([false]);
});

test('should emit change event when click the switch button', async () => {
  const wrapper = mount(Switch);

  wrapper.trigger('click');
  expect(wrapper.emitted('change')).toHaveLength(1);
  expect(wrapper.emitted('change')![0]).toEqual([true]);

  await wrapper.setProps({ modelValue: true });
  wrapper.trigger('click');
  expect(wrapper.emitted('change')).toHaveLength(2);
  expect(wrapper.emitted('change')![1]).toEqual([false]);
});

test('should not emit change event or update:modelValue event if disabled', async () => {
  const wrapper = mount(Switch, {
    props: {
      disabled: true,
    },
  });

  wrapper.trigger('click');
  expect(wrapper.emitted('change')).toBeFalsy();
  expect(wrapper.emitted('update:modelValue')).toBeFalsy();
});

test('should change active color when using active-color prop', () => {
  const wrapper = mount(Switch, {
    props: {
      modelValue: true,
      activeColor: 'black',
    },
  });

  expect(wrapper.style.backgroundColor).toEqual('black');
});

test('should change inactive color when using inactive-color prop', () => {
  const wrapper = mount(Switch, {
    props: {
      inactiveColor: 'black',
    },
  });

  expect(wrapper.style.backgroundColor).toEqual('black');
});

test('should apply active color to loading icon', () => {
  const wrapper = mount(Switch, {
    props: {
      loading: true,
      modelValue: true,
      activeColor: 'red',
    },
  });

  const loading = wrapper.find('.van-switch__loading');
  expect(loading.html()).toMatchSnapshot();
});

test('should apply inactive color to loading icon', () => {
  const wrapper = mount(Switch, {
    props: {
      loading: true,
      inactiveColor: 'black',
    },
  });

  const loading = wrapper.find('.van-switch__loading');
  expect(loading.html()).toMatchSnapshot();
});

test('should change size when using size prop', () => {
  const wrapper = mount(Switch, {
    props: {
      size: 20,
    },
  });

  expect(wrapper.style.fontSize).toEqual('20px');
});

test('should allow to custom active-value and inactive-value', () => {
  const wrapper = mount(Switch, {
    props: {
      modelValue: 'on',
      activeValue: 'on',
      inactiveValue: 'off',
    },
  });

  expect(wrapper.find('.van-switch--on').exists()).toBeTruthy();

  wrapper.trigger('click');
  expect(wrapper.emitted('update:modelValue')![0]).toEqual(['off']);
});

test('should render node slot correctly', () => {
  const wrapper = mount(Switch, {
    slots: {
      node: () => 'Custom node',
    },
  });

  expect(wrapper.html()).toMatchSnapshot();
});

test('should render background slot correctly', () => {
  const wrapper = mount(Switch, {
    slots: {
      background: () => 'Custom background',
    },
  });

  expect(wrapper.html()).toMatchSnapshot();
});
