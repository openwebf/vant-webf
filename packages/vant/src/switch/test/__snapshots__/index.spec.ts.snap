// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should apply active color to loading icon 1`] = `
<div class="van-loading van-loading--circular van-switch__loading"
     aria-live="polite"
     aria-busy="true"
>
  <span class="van-loading__spinner van-loading__spinner--circular"
        style="color: red;"
  >
    <svg class="van-loading__circular"
         viewbox="25 25 50 50"
    >
      <circle cx="50"
              cy="50"
              r="20"
              fill="none"
      >
      </circle>
    </svg>
  </span>
</div>
`;

exports[`should apply inactive color to loading icon 1`] = `
<div class="van-loading van-loading--circular van-switch__loading"
     aria-live="polite"
     aria-busy="true"
>
  <span class="van-loading__spinner van-loading__spinner--circular"
        style="color: black;"
  >
    <svg class="van-loading__circular"
         viewbox="25 25 50 50"
    >
      <circle cx="50"
              cy="50"
              r="20"
              fill="none"
      >
      </circle>
    </svg>
  </span>
</div>
`;

exports[`should render background slot correctly 1`] = `
<div role="switch"
     class="van-switch"
     tabindex="0"
     aria-checked="false"
>
  <div class="van-switch__node">
  </div>
  Custom background
</div>
`;

exports[`should render node slot correctly 1`] = `
<div role="switch"
     class="van-switch"
     tabindex="0"
     aria-checked="false"
>
  <div class="van-switch__node">
    Custom node
  </div>
</div>
`;
